In [1]:
import numpy as np
import matplotlib.pyplot as plt

In [2]:
m=4
N = 50
d = 5
L = d+(d*m)*2

T = np.zeros(5*N)
x = np.linspace(0,L,5*N)


In [3]:
def Tfunc(T):
    for i in range(len(T)):
        if x[i]>10 and x[i]<=15:
            T[i]= 1230+273 # 1150 in Kelvin
        else:
            T[i]=20+273 # 0 Deg in Kelvin
    return T

T = Tfunc(T)
#print(T)


In [4]:
k = 5.2 #J/m/s/K
c = 1300 #J/kg/K
p = 2750 #kg/m^3
delta_t = 1800//(m**2) #sec
delta_x = (L/(N*5))**2
n = (delta_t/delta_x)*(k/(c*p)) 

print("n:", n)

def tchange(T,n):
    Tnew=np.copy(T)
    for i in range(1,N*5-1):
        Tnew[i] = n*((T[i+1])-(2*T[i])+T[i-1])+(T[i])
        Tnew = T
    return T

#print(tchange(T,n))

n: 0.005028058361391695


In [5]:
t_org = 1800

hour = 60*60 
day = 60*60*24
week = 60*60*24*7
month = 60*60*24*30
year = 60*60*24*365

time = np.array([hour , day, week, month, year])
timecalc = time//delta_t

for t in timecalc:
    T = Tfunc(T)
    for j in range(t):
        T=tchange(T,n)
    print(max(T))

1503.0
1502.9955274519475
1432.119975317856
1067.4549119905673
534.5878157804455


In [9]:
#6 multiplier

m=6
N = 50
d = 5
L = d+(d*m)*2

T = np.zeros(5*N)
x = np.linspace(0,L,5*N)

In [10]:
def Tfunc(T):
    for i in range(len(T)):
        if x[i]>10 and x[i]<=15:
            T[i]= 1230+273 # 1150 in Kelvin
        else:
            T[i]=20+273 # 0 Deg in Kelvin
    return T

T = Tfunc(T)
#print(T)


In [11]:
k = 5.2 #J/m/s/K
c = 1300 #J/kg/K
p = 2750 #kg/m^3
delta_t = 1800//(m**2) #sec
delta_x = (L/(N*5))**2
n = (delta_t/delta_x)*(k/(c*p)) 

print("n:", n)

def tchange(T,n):
    Tnew=np.copy(T)
    for i in range(1,N*5-1):
        Tnew[i] = n*((T[i+1])-(2*T[i])+T[i-1])+(T[i])
        Tnew = T
    return T

#print(tchange(T,n))

n: 0.0010758472296933835


In [13]:
t_org = 1800

hour = 60*60 
day = 60*60*24
week = 60*60*24*7
month = 60*60*24*30
year = 60*60*24*365

time = np.array([hour , day, week, month, year])
timecalc = time//delta_t

for t in timecalc:
    T = Tfunc(T)
    for j in range(t):
        T=tchange(T,n)
    print(max(T))

1503.0
1502.9869008443343
1427.3547380516504
1057.63178334145
530.0219568647199


In [14]:
#6 multiplier

m=8
N = 50
d = 5
L = d+(d*m)*2

T = np.zeros(5*N)
x = np.linspace(0,L,5*N)

In [15]:
def Tfunc(T):
    for i in range(len(T)):
        if x[i]>10 and x[i]<=15:
            T[i]= 1230+273 # 1150 in Kelvin
        else:
            T[i]=20+273 # 0 Deg in Kelvin
    return T

T = Tfunc(T)
#print(T)


In [16]:
k = 5.2 #J/m/s/K
c = 1300 #J/kg/K
p = 2750 #kg/m^3
delta_t = 1800//(m**2) #sec
delta_x = (L/(N*5))**2
n = (delta_t/delta_x)*(k/(c*p)) 

print("n:", n)

def tchange(T,n):
    Tnew=np.copy(T)
    for i in range(1,N*5-1):
        Tnew[i] = n*((T[i+1])-(2*T[i])+T[i-1])+(T[i])
        Tnew = T
    return T

#print(tchange(T,n))

n: 0.0010758472296933835


In [17]:

hour = 60*60 
day = 60*60*24
week = 60*60*24*7
month = 60*60*24*30
year = 60*60*24*365

time = np.array([hour , day, week, month, year])
timecalc = time//delta_t

for t in timecalc:
    T = Tfunc(T)
    for j in range(t):
        T=tchange(T,n)
    print(max(T))
    

1503.0
1502.9869008443343
1427.3547380516504
1057.63178334145
530.0219568647199


### Question 3

| cooling time | CONVERGED d*8 Temperature at the dike center (K) |
| --- | --- |
| 1 hour | 1503 |
| 1 day | 1503 |
| 1 week | 1427 |
| 1 month | 1057 |
| 1 year |  530  |